lib/etcdv3.rb

```diff
@@ -82,101 +84,101 @@ def get(key, opts={})
   end
 
   # Inserts a new key.
-  def put(key, value, lease_id: nil)
-    @conn.handle(:kv, 'put', [key, value, lease_id])
+  def put(key, value, opts={})
+    @conn.handle(:kv, 'put', [key, value, opts])
   end
 
   # Deletes a specified key
-  def del(key, range_end: '')
-    @conn.handle(:kv, 'del', [key, range_end])
+  def del(key, opts={})
+    @conn.handle(:kv, 'del', [key, opts])
   end
 
   # Grant a lease with a specified TTL
-  def lease_grant(ttl)
-    @conn.handle(:lease, 'lease_grant', [ttl])
+  def lease_grant(ttl, opts={})
+    @conn.handle(:lease, 'lease_grant', [ttl, opts])
   end
 
   # Revokes lease and delete all attached keys
-  def lease_revoke(id)
-    @conn.handle(:lease, 'lease_revoke', [id])
+  def lease_revoke(id, opts={})
+    @conn.handle(:lease, 'lease_revoke', [id, opts])
   end
 
   # Returns information regarding the current state of the lease
-  def lease_ttl(id)
-    @conn.handle(:lease, 'lease_ttl', [id])
+  def lease_ttl(id, opts={})
+    @conn.handle(:lease, 'lease_ttl', [id, opts])
   end
 
   # List all roles.
-  def role_list
-    @conn.handle(:auth, 'role_list')
+  def role_list(opts={})
+    @conn.handle(:auth, 'role_list', [opts])
   end
 
   # Add role with specified name.
-  def role_add(name)
-    @conn.handle(:auth, 'role_add', [name])
+  def role_add(name, opts={})
+    @conn.handle(:auth, 'role_add', [name, opts])
   end
 
   # Fetches a specified role.
-  def role_get(name)
-    @conn.handle(:auth, 'role_get', [name])
+  def role_get(name, opts={})
+    @conn.handle(:auth, 'role_get', [name, opts])
   end
 
   # Delete role.
-  def role_delete(name)
-    @conn.handle(:auth, 'role_delete', [name])
+  def role_delete(name, opts={})
+    @conn.handle(:auth, 'role_delete', [name, opts])
   end
 
   # Grants a new permission to an existing role.
-  def role_grant_permission(name, permission, key, range_end='')
-    @conn.handle(:auth, 'role_grant_permission', [name, permission, key, range_end])
+  def role_grant_permission(name, permission, key, range_end='', opts={})
+    @conn.handle(:auth, 'role_grant_permission', [name, permission, key, range_end, opts])
   end
 
-  def role_revoke_permission(name, permission, key, range_end='')
-    @conn.handle(:auth, 'role_revoke_permission', [name, permission, key, range_end])
+  def role_revoke_permission(name, permission, key, range_end='', opts={})
+    @conn.handle(:auth, 'role_revoke_permission', [name, permission, key, range_end, opts])
   end
 
   # Fetch specified user
-  def user_get(user)
-    @conn.handle(:auth, 'user_get', [user])
+  def user_get(user, opts={})
+    @conn.handle(:auth, 'user_get', [user, opts])
   end
 
   # Creates new user.
-  def user_add(user, password)
-    @conn.handle(:auth, 'user_add', [user, password])
+  def user_add(user, password, opts={})
```

Comment:
Given these methods are directly accessed by the user, we are going to want these to be as clear as possible.   Users, for the most part, should know exactly what can and cannot be passed into these methods by looking at this file.  

With that being said, where a method doesn't allow multiple options we should just specify `timeout` instead of `opts` so it's clear that there's not additional options that can be specified.  If opts makes the most sense like the case of the `get` method, then we need to document the available options.  

I hope that make sense!